File: 151-reverse-words-in-a-string/reverse-words-in-a-string.cpp
class Solution {
public:
    string reverseWords(string s) {
      int i = s.length() - 1;
    string ans = "";

    while (i >= 0)
    {
        while (i >= 0 && s[i] == ' ')
            i--;
        int j = i;
        if (i < 0)
            break;
        while (i >= 0 && s[i] != ' ')
            i--;
        if (ans.empty())
        {
            ans = s.substr(i + 1, j - i);
        }
        else
        {
            ans = ans + " " + s.substr(i + 1, j - i);
        }
    }
    return ans;
    }
};

File: 151-reverse-words-in-a-string/reverse-words-in-a-string.java
    class Solution {
    public String reverseWords(String s) {
        String a[] = s.split("\\s+");
        String rev ="";
        for (int i = a.length - 1; i >= 0; i--) {
            rev += a[i]+" ";
        }
        return rev.trim();
    }
};